code/game/mecha/equipment/tools/tools.dm

```diff
@@ -1417,5 +1417,54 @@
 	else
 		T.ChangeTurf(/turf/simulated/floor/plating)
 
+/obj/item/mecha_parts/mecha_equipment/tool/collector
+	name = "\improper Exosuit-Mounted Radiation Collector Array"
+	desc = "An exosuit-mounted Radiation Collector Array. (Can be attached to: Any exosuit)"
+	icon_state = "mecha_rcd"	//needtochange
+	origin_tech = Tc_PLASMATECH + "=3;" + Tc_MAGNETS + "=2;" + Tc_POWERSTORAGE + "=4"
+	equip_cooldown = 10
+	energy_drain = 0
+	range = MELEE
+	var/active = FALSE
+	var/obj/machinery/power/rad_collector/mech/collector
+
+/obj/item/mecha_parts/mecha_equipment/tool/collector/New()
+	..()
+	collector = new(src)
+	collector.connected_module = src
+
+/obj/item/mecha_parts/mecha_equipment/tool/collector/Destroy()
+	collector.connected_module = null
+	qdel(collector)
+	collector = null
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/tool/collector/action(atom/target)
+	var/obj/item/weapon/tank/plasma/plas = target
+	if(istype(plas))
+		if(collector.P)
+			occupant_message("There is already a tank in the radiation collector array.")
+			return
+		plas.forceMove(collector)
+		collector.P = target
+		occupant_message("You insert \the [target] into the radiation collector array.")
+		update_equip_info()
+
+/obj/item/mecha_parts/mecha_equipment/tool/collector/Topic(href,href_list)
+	..()
+	if(href_list["toggle"])
```

Comment:
href sanity!

Reply:
You're going to have to tell me what to do specifically, I don't often work with hrefs.

Reply:
Make it ensure the user is conscious and the occupant of the mech. Preferably inside the super call.

Reply:
I imagine I should do that for all the `Topic()`s?

Reply:
Yes.